src/query_plan/fetch_dependency_graph.rs

```diff
@@ -140,21 +169,22 @@ pub(crate) struct FetchDependencyGraph {
 }
 
 // TODO: Write docstrings
-#[derive(Debug, Clone)]
+#[derive(Debug)]
 pub(crate) struct DeferTracking {
-    pub(crate) top_level_deferred: IndexSet<NodeStr>,
-    pub(crate) deferred: IndexMap<NodeStr, DeferredInfo>,
-    pub(crate) primary_selection: Option<Arc<NormalizedSelectionSet>>,
```

Comment:
We should really only use `Arc` when we can't guarantee the lifetime (or when doing so would be inconvenient), and `Clone` derives when absolutely necessary, so I remove some of the eager Arcs and Clones in this PR.